### PR TITLE
chore(docs): Update README to Include Async Ex

### LINKED ISSUE #6: proposal: Update README Examples
Hi there team!

Your package here is stupendous and valuable, so thank you for your hard work! This topic is, somehow, not super easy for some reason. You'd think, the browser should just do its thing and ✅ ? 

A common use case that I ran into is waiting for a specific element to render before `updateScroll` is called. I read over the README and I read over your tests (thanks for adding those!). Unfortunately, I couldn't quite piece together just "how" registering an element would allow me to "wait for it to render" before updating scroll. 

So what I did instead was use the `update ref` pattern that React [outlines here](https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node). 

My implementation ended up looking a little like this:

```ts
const MyComponent = () => {
  const divRef = useCallback(
    (node) => {
      if (node !== null) {
        updateScroll();
      }
    },
    [updateScroll]
  );

  return (
    <div ref={divRef}>hi</div>
  );
}
```

This ended up working pretty well for me, and allowed me to explicitly wait for a node to appear before updating scroll. 

The reason I'm opening this issue, is I'm wondering if the README could be more clear about how to use `registerElement` and `updateScroll` together. If the intended use case for those is not similar to the above implementation, would it be valuable to add an example that showed how to do something like this? 

Thanks again for all your hard work!

### PR DESCRIPTION
**Pull request by therynamo**
Most complex UIs will utilize, in one form or another, async rendering. It is helpful to see how this library may be useful in approaching that problem.

closes #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,6 +138,28 @@ Call `updateScroll` function whenever you want to update the scroll. You may opt
...
+```ts
+const MyComponent = () => {
+  const divRef = useCallback(
+    (node) => {
+      if (node !== null) {
+        updateScroll();
+      }
+    },
+    [updateScroll]
+  );
```
Comment (satazor) [inline README.md:154]:
```suggestion
    const divRef = useCallback((node) => {
        if (node) {
            updateScroll();
        }
    }, [updateScroll]
);
```

File: README.md [hunk 2]
```diff
@@ -138,6 +138,28 @@ Call `updateScroll` function whenever you want to update the scroll. You may opt
...
+
+  // const someAsyncData = await Promise(/*...*/);
+  return (
+    {someAsyncData && <div ref={divRef}>hi</div>}
+  );
+}
```
Comment (satazor) [inline README.md:160]:
```suggestion

    return someCondition ? <div ref={ divRef }>hi</div> : null;
};
```

**Review by satazor — CHANGES_REQUESTED** [review]
Just some minor changes based on the code style of the rest of the README.

**Comment by therynamo** [comment]
Perfect! Updated @satazor

**Pull request merged**